earth_enterprise/src/common/FileAccessor.cpp

```diff
@@ -0,0 +1,35 @@
+#include "FileAccessor.h"
+
+// ****************************************************************************
+// ***  POSIXFileAccessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, int flags, mode_t createMask) {
+  return khOpen(fname, flags, createMask);
+}
+
+int POSIXFileAccessor::FsyncAndClose() {
+  return khFsyncAndClose(this->getAsFD());
+}
+
+int POSIXFileAccessor::Close() {
+  return khClose(this->getAsFD());
+}
+
+bool POSIXFileAccessor::PreadAll(void* buffer, size_t size, off64_t offset) {
+  return khPreadAll(this->getAsFD(), buffer, size, offset);
+}
+
+bool POSIXFileAccessor::PwriteAll(const void* buffer, size_t size, off64_t offset) {
+  return khPwriteAll(this->getAsFD(), buffer, size, offset);
+}
+
+std::unique_ptr<AbstractFileAccessor> AbstractFileAccessor::getAccessor(const std::string &fname, int flags, mode_t createMask) {
+  //notify(NFY_WARN, "Filename: %s", fname.c_str());
+  if (fname.rfind("/gevol", 0) == 0 || fname.rfind("/tmp", 0) == 0) {
```

Comment:
I think this will miss cases where the asset root or publish root are not under `/gevol`. That's the default, but users can specify different asset and publish roots. I _think_ it might be enough just to see if the path starts with `/`.
@tst-lsavoie Thoughts?

Reply:
I'd just use a `POSIXFileAccessor` here by default. We can worry about picking the right accessor when we implement the plugin capability.